thrvrce/reactNative
# Task7

TASKS 1

1. Create ‘dev’, ‘qa’ and ‘prod’ flavors
- сделано
 ![image](https://user-images.githubusercontent.com/70852808/167268044-94fe986e-5ede-4730-a93e-c3b79f8364be.png)
2. Create groups on Appcenter project for qa and prod build.
- сделано 
![image](https://user-images.githubusercontent.com/70852808/167268085-a3870933-0962-4493-915b-56ce041ae32c.png)
3. Use react-native-dotenv(react native-config) for storing Appcenter’s access tokens and groups names.
- не понял для чего это надо. Access token для app Center нужен только в джобах для экшенов. Для имен групп тоже не понял что за имена и для чего их надо получать из переменных окружения.
4. Create pipeline which run lint and unit tests for each commit to any branch.
- готово. см. файл testsAndLinterJobs.yml
![image](https://user-images.githubusercontent.com/70852808/167268680-5f81a8c9-5c46-4f5a-9086-98215aa71762.png)

TASKS 2

1. Create pipeline for develop branch which build Qa flavor app and pushes it to Appcenter Qa group.
- сделано. см. файл android_build_QA.yml
![image](https://user-images.githubusercontent.com/70852808/167268657-fddb8bf3-134b-43b0-ae37-24860096d202.png)

2. Create pipeline for “master” branch. It should build Prod flavor by each commit and deploy it to Appcenter.
- не уверен что сделано, т.к. не понял что значит "deploy it to appcenter'. Сделал билд Prod flavor и пуш .apk в Prod группу.
![image](https://user-images.githubusercontent.com/70852808/167268636-a16cd670-134a-44f8-a4af-cfef9353bd2b.png)

TASKS 3

1. Create pipeline for release branch. Branch format is “release-vX.Y” where X and Y are numbers. Pipeline should trigger after creating release branch and by each commit in it and deploy artifact to Appcenter `prod` group.
- готово. См. файл android_build_Release.yml
![image](https://user-images.githubusercontent.com/70852808/167268421-ee04616d-4ae6-4a41-ad94-4a2568e4e26a.png)

2. (optional) move Appcenter tokens from .env file to env variable of CI tool and apply it on a build. Appcenter token should be stored only in CI environment.
- сделано. Используется секрет secrets.APP_CENTER_TOKEN
